packages/ui/Scrollable/Scrollable.js

```diff
@@ -136,6 +136,19 @@ class ScrollableBase extends Component {
 		 */
 		clearOverscrollEffect: PropTypes.func,
 
+		/**
+		 * Client size of the container; valid values are an object that has `clientWidth` and `clientHeight`.
+		 *
+		 * @type {Object}
+		 * @property {Number}    clientHeight    The client height of the list.
+		 * @property {Number}    clientWidth    The client width of the list.
+		 * @public
+		 */
+		clientSize: PropTypes.shape({
+			clientHeight: PropTypes.number.isRequired,
```

Comment:
Is it really necessary to supply both values if your container is only scrolling in one direction?  Should we elaborate that this is an opt-in feature and why you'd want to use it?

Reply:
I think you make a good point but I'm not sure we want to muck with the API right now since we'd want to change VirtualList to match this. We could make it backwards compatible by supporting either a single value or the object but would then add complexity to handle it internally. Guess it depends on how important you feel that DX change is.

Adding docs (here and in VL) makes sense though.

Reply:
I think simply removing the `isRequired` would be sufficient.  Don't need to change it to accept a single value vs. object, imo.

Reply:
Hm.. probably you are right for scrollbar's perspective but VL needs both for calculating item size. I am not confident it will work only with `clientHeight` or `clientWidth`. Our intention was `clientSize` should have both for saving render cycle, or it will be OK without `clientSize` if the perf. is not too bad

Reply:
`clientSize` is to specify the dimension of the container's viewport essentially, so it should require both values. I understand it's probably unnecessary for `Scrollbar` to have both values as @seunghoh mentioned, but `Scrollable` is the parent component of both `Scrollbar` and `VirtualList` so it should require both values here.

On a side note, these are all private components, so I suggest we revisit later if you're concerned with APIs. As long as we agree on the fact that it gives a performance boost, I say go with it.